### PR TITLE
Refactor of the searching and searching labels

### PR DESCRIPTION
**Pull request by eanker**
The code for searching is now scalable.
We can now also search in the labels of a pull request.

### CODE REVIEW

File: src/projects/project-pull-requests.html [hunk 1]
```diff
@@ -125,16 +125,30 @@
         hideProjectButton: {
           type: String,
           value: ''
+        },
+        getFields: {
+          type: Function,
+          value: function() {
+            return function(e) {
```
Comment (txsmith) [inline src/projects/project-pull-requests.html]: I think the argument name could be made more explicit to reflect that it is a pull request object.

File: src/projects/projects-page.html [hunk 2]
```diff
@@ -162,6 +162,14 @@
               }
             ];
           }
+        },
+        getFields: {
+          type: Function,
+          value: function() {
+            return function(e) {
```
Comment (txsmith) [inline src/projects/projects-page.html]: Same goes for this argument.

File: src/projects/project-pull-requests.html (hunk 1 above)
Comment (eanker) [inline src/projects/project-pull-requests.html]: Would getSearchFields be better?
The name should be the same for search-behaviour projects and pr's.

File: src/projects/search-behavior.html [hunk 3]
```diff
@@ -1,24 +1,37 @@
...
+      else {
+        searchWord = searchWord.toLowerCase();
+        return list.filter(function(e) {
+          var count = 0;
+          getFields(e).forEach(function(f){
+            if (f.toLowerCase().indexOf(searchWord) > -1) {
+              count++;
+            }
+          });
+          return count > 0;
```
Comment (txsmith) [inline src/projects/search-behavior.html]: This count is unnecessary because the for-loop can exit as soon as one match is found.

Comment (txsmith) [inline src/projects/search-behavior.html]: However, returning from a for-loop does not work with foEach, you should use a normal indexed loop for this instead.

File: src/projects/project-pull-requests.html (hunk 1 above)
Comment (txsmith) [inline src/projects/project-pull-requests.html]: My comment was aimed at the `e` argument, not `getFields`

**Comment by eanker** [comment]
I have fixed your comments (and rebased).

**Pull request merged**